sdk/appconfiguration/azure-appconfiguration-provider/CHANGELOG.md

```diff
@@ -5,10 +5,14 @@
 ### Features Added
 
 ### Breaking Changes
+* Added `AzureAppConfigurationKeyVaultOptions` to take in a `client_configs` a Mapping of endpoints to client kwargs instead of taking in the whole client.
+* Removed `AzureAppConfigurationKeyVaultOptions` `secret_clients`, `client_configs` should be used instead.
```

Comment:
One thing that struck me (after the review yesterday) is that there isn't a clear way to specify the default options (e.g. the options to be used for all secret clients). Not sure if that is needed?

Reply:
(and with some trickery, a persistent enough user can use a `DefaultDict` as long as clients are on-demand created and one doesn't have to enumerate keys)

Reply:
I was thinking about this too. Without expanding the API too much we could allow a "default" key, with a dictionary. Thoughts. I don't fell forcing DefaultDict as the right option.

Reply:
Thinking. One option is to rather than taking a `dict` taking a list of configurations with an optional vault url/prefix. And check them in order. I'm torn between "this is an advanced scenario, so making it trivial to use is less of a priority than covering edge cases" and "this is annoyingly complicated".

How often do you expect people to have different options per vault vs. the same options?

Reply:
We have no metrics on it. But I would guess that they are more likely to be the same than different based on use cases I have seen in the past.

I am not sure what you mean by "check them in order", unless you just mean iterate through the list every time.

Reply:
Staying as is for now.